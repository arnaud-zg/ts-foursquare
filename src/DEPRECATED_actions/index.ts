import { TLifeAction } from './life'

export type TRootAction = TLifeAction

export * from './life'
