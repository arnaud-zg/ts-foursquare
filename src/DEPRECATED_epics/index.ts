export * from './life'
export * from './lists'
export * from './photos'
export * from './venues'
