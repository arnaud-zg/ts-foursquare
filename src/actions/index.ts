import { getLists } from './lists'
import { getPhotosDetails } from './photos'

export const actions = { getLists, getPhotosDetails }

export type TActions = typeof actions
export type TActionsValue = TActions[keyof TActions]
