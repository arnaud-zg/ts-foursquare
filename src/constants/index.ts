export const PROJECT_NAME = 'ts-foursquare'
