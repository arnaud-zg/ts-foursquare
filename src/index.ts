require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
