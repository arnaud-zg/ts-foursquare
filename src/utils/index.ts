export * from './icon'
export * from './url'
