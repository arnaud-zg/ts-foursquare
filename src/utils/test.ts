import { NHttpStatuses } from 'http-response-status'

export const mockingFetch = ({ response = {} } = {}) => {
  const mockJsonPromise = Promise.resolve(response)
  const mockFetchPromise = Promise.resolve({
    json: () => mockJsonPromise,
    ok: true,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}

export const mockingFailingNotValidFetch = () => {
  const mockFetchPromise = Promise.resolve({
    ok: true,
    json: null,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}

export const mockingFailingNotFoundFetch = () => {
  const mockFetchPromise = Promise.resolve({
    ok: false,
    status: NHttpStatuses.EClientError.NOT_FOUND,
  })

  // @ts-ignore
  global.fetch = jest.fn().mockImplementation(() => mockFetchPromise)
}
