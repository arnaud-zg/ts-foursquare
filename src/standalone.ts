import { actions } from './actions'

export enum EReturnType {
  OBSERVABLE = 'OBSERVABLE',
  PROMISE = 'PROMISE',
}

export interface IConfigParams {
  config: IStandaloneConfig
}

export interface IStandaloneConfig {
  clientId: string
  clientSecret: string
  accessToken?: string
  returnType?: EReturnType
}

export class tsFoursquare {
  getLists: ReturnType<typeof actions.getLists>
  getPhotosDetails: ReturnType<typeof actions.getPhotosDetails>
  getVenuesCategories: ReturnType<typeof actions.getVenuesCategories>
  getVenuesExplore: ReturnType<typeof actions.getVenuesExplore>
  getVenuesLikes: ReturnType<typeof actions.getVenuesLikes>
  getVenuesListed: ReturnType<typeof actions.getVenuesListed>
  getVenuesNextVenues: ReturnType<typeof actions.getVenuesNextVenues>
  getVenuesSearch: ReturnType<typeof actions.getVenuesSearch>
  getVenuesSimilar: ReturnType<typeof actions.getVenuesSimilar>
  getVenuesSuggestCompletion: ReturnType<
    typeof actions.getVenuesSuggestCompletion
  >
  getVenuesTrending: ReturnType<typeof actions.getVenuesTrending>

  constructor({
    clientId,
    clientSecret,
    returnType = EReturnType.PROMISE,
  }: IStandaloneConfig) {
    const config = {
      clientId,
      clientSecret,
      returnType,
    }

    this.getLists = actions.getLists({ config })
    this.getPhotosDetails = actions.getPhotosDetails({ config })
    this.getVenuesCategories = actions.getVenuesCategories({ config })
    this.getVenuesExplore = actions.getVenuesExplore({ config })
    this.getVenuesLikes = actions.getVenuesLikes({ config })
    this.getVenuesListed = actions.getVenuesListed({ config })
    this.getVenuesNextVenues = actions.getVenuesNextVenues({ config })
    this.getVenuesSearch = actions.getVenuesSearch({ config })
    this.getVenuesSimilar = actions.getVenuesSimilar({ config })
    this.getVenuesSuggestCompletion = actions.getVenuesSuggestCompletion({
      config,
    })
    this.getVenuesTrending = actions.getVenuesTrending({ config })
  }
}
