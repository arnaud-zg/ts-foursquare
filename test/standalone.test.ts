import { tsFoursquare } from '../src/standalone'
import { mockingFetch } from '../src/utils/test'
import { payload as payloadGetLists } from './actions/__mocks__/getListsAsync.resolve'

describe('tsFoursquare', () => {
  it('should get a snapshot of tsFoursquare instance', () => {
    mockingFetch({ response: { ...payloadGetLists } })
    const ts4Sq = new tsFoursquare({ clientId: '', clientSecret: '' })
    ts4Sq.actions.getLists({
      payload: { listId: '5580721e498e7c48540bf83f' },
    })
    expect(ts4Sq).toMatchSnapshot()
  })
})
