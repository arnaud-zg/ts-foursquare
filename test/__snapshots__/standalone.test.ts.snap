// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tsFoursquare should get a snapshot of tsFoursquare instance 1`] = `
tsFoursquare {
  "actions": Object {
    "getLists": [Function],
    "getPhotosDetails": [Function],
  },
  "clientId": "",
  "clientSecret": "",
  "injectContextInAction": [Function],
  "returnType": "PROMISE",
}
`;
